src/main/java/org/generationcp/middleware/api/location/search/LocationSearchRequest.java

```diff
@@ -12,28 +12,30 @@
 public class LocationSearchRequest {
 
 	private String programUUID;
-	private Set<Integer> locationTypes = new HashSet<>();
+	private Set<Integer> locationTypeIds = new HashSet<>();
 	private List<Integer> locationIds = new ArrayList<>();
 	private List<String> locationAbbreviations = new ArrayList<>();
+	private String locationId;
```

Comment:
You don't need separate fields for a single locationId and single locationType. You can create singleton set and use in  existing `locationTypeIds` and `locationIds` fields.